src/OpenLoco/src/GameCommands/Airports/RemoveAirport.cpp

```diff
@@ -0,0 +1,275 @@
+#include "RemoveAirport.h"
+#include "Economy/Economy.h"
+#include "Localisation/StringIds.h"
+#include "Map/StationElement.h"
+#include "Map/SurfaceElement.h"
+#include "Map/TileElement.h"
+#include "Map/TileManager.h"
+#include "Map/Track/TrackData.h"
+#include "Objects/AirportObject.h"
+#include "Objects/ObjectManager.h"
+#include "Vehicles/Vehicle.h"
+#include "Vehicles/VehicleManager.h"
+#include "ViewportManager.h"
+#include "World/Industry.h"
+#include "World/Station.h"
+#include "World/StationManager.h"
+#include "World/TownManager.h"
+
+namespace OpenLoco::GameCommands
+{
+    constexpr std::array<World::Pos2, 4> _largeTileOffsets = { {
+        { 0, 0 },
+        { 0, -32 },
+        { -32, -32 },
+        { -32, 0 },
+    } };
+
+    static World::StationElement* getStationEl(const World::Pos3& pos)
+    {
+        auto tile = World::TileManager::get(pos);
+        for (auto& el : tile)
+        {
+            auto* stationEl = el.as<World::StationElement>();
+            if (stationEl == nullptr)
+            {
+                continue;
+            }
+
+            if (stationEl->baseHeight() == pos.z)
+            {
+                return stationEl;
+            }
+        }
+
+        return nullptr;
+    }
+
+    // 0x004938D9
+    static bool removeAirportTileElement(const World::Pos3& pos, const AirportObject* airportObj, const uint8_t bh, const uint8_t flags)
+    {
+        for (auto& searchTile : getBuildingTileOffsets(airportObj->largeTiles & (1 << bh)))
+        {
+            const auto portPos = World::Pos3(searchTile.pos + pos, pos.z);
```

Comment:
port?

Reply:
Haha, well, an airport is a port, isn't it 😄 I'll rename it to airportPos.